autoscale/src/main/java/com/sequenceiq/periscope/service/ClusterService.java

```diff
@@ -168,10 +173,17 @@ public Cluster findById(Long clusterId) {
     public void removeById(Long clusterId) {
         Cluster cluster = findById(clusterId);
         LoggingUtils.buildMdcContext(cluster);
+        clearScalingActivityForCluster(clusterId);
         clusterRepository.delete(cluster);
         calculateClusterStateMetrics();
     }
 
+    private void clearScalingActivityForCluster(Long clusterId) {
```

Comment:
Why do we need to load the ids into memory, and then issue a delete?

Shouldn't this be a `delete from scaling_activity` where cluster_id=`${clusterId}`